File: references/MWORKS高校星火计划资料包/培训课程配套材料/01-官网课程配套材料/00-快速入门课程/02-Syslab快速入门/05-Julia语法-入门/附件示例/对应课程PPT页码/07_与其他语言的差异/07-2Julia与Python语言的显著差异/page_284/page_284.jl
# 关键词参数在 ; 之后
function mul(; x, y, kwargs...)
  z = 0
  if haskey(kwargs, :z)
    z = kwargs[:z]
  end
  return x + 2 * y + z
end
mul(y=2, x=1, z=1)
mul(y=2, x=1, z=1)
# 1 + 2*2 + 1 = 6

File: References/MWORKS/MWORKS高校星火计划资料包/培训课程配套材料/01-官网课程配套材料/00-快速入门课程/02-Syslab快速入门/05-Julia语法-入门/附件示例/对应课程PPT页码/07_与其他语言的差异/07-2Julia与Python语言的显著差异/page_284/page_284.py
# 不定长的关键词参数使用 **kwargs 表示
def mul(x, y, **kwargs):
  z = 0
  if "z" in kwargs:
    z = kwargs["z"]

  return x + 2 * y + z

mul(y = 2, x = 1, z = 1)
# 2 + 1*2 + 1 = 6




